helm-chart/templates/secrets.yaml

```diff
@@ -0,0 +1,125 @@
+---
+{{- if .Values.secrets.localSettings.DATABASES_HOST }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: local-settings-secrets-{{ .Values.deployEnv }}
+  labels:
+    deployEnv: "{{ .Values.deployEnv }}"
+type: Opaque
+data:
+  {{- with .Values.secrets.localSettings }}
+  DATABASES_HOST: {{ .DATABASES_HOST | b64enc }}
+  DATABASES_PASS: {{ .DATABASES_PASS | b64enc }}
+  DATABASES_USER: {{ .DATABASES_USER | b64enc }}
+  DATABASES_PORT: {{ .DATABASES_PORT | b64enc }}
+  GOOGLE_OAUTH2_CLIENT_ID: {{ .GOOGLE_OAUTH2_CLIENT_ID | b64enc }}
+  GOOGLE_OAUTH2_CLIENT_SECRET: {{ .GOOGLE_OAUTH2_CLIENT_SECRET | b64enc }}
+  SECRET_KEY: {{ .SECRET_KEY | b64enc }}
+  MANDRILL_API_KEY: {{ .MANDRILL_API_KEY | b64enc }}
+  SEFARIA_DB: {{ .SEFARIA_DB | b64enc }}
+  SEFARIA_DB_USER: {{ .SEFARIA_DB_USER | b64enc }}
+  SEFARIA_DB_PASSWORD: {{ .SEFARIA_DB_PASSWORD | b64enc }}
+  SEARCH_ADMIN_USER: {{ .SEARCH_ADMIN_USER | b64enc }}
+  SEARCH_ADMIN_PW: {{ .SEARCH_ADMIN_PW | b64enc }}
+  SEARCH_ADMIN_K8S: {{ .SEARCH_ADMIN_K8S | b64enc }}
+  TURN_SECRET: {{ .TURN_SECRET | b64enc }}
+  TURN_USER: {{ .TURN_USER | b64enc }}
+  SEFARIA_BOT_API_KEY: {{ .SEFARIA_BOT_API_KEY | b64enc }}
+  CLOUDFLARE_ZONE: {{ .CLOUDFLARE_ZONE | b64enc }}
+  CLOUDFLARE_EMAIL: {{ .CLOUDFLARE_EMAIL | b64enc }}
+  CLOUDFLARE_TOKEN: {{ .CLOUDFLARE_TOKEN | b64enc }}
+  GOOGLE_TAG_MANAGER_CODE: {{ .GOOGLE_TAG_MANAGER_CODE | b64enc }}
+  GOOGLE_ANALYTICS_CODE: {{ .GOOGLE_ANALYTICS_CODE | b64enc }}
+  GOOGLE_MAPS_API_KEY: {{ .GOOGLE_MAPS_API_KEY | b64enc }}
+  MIXPANEL_CODE: {{ .MIXPANEL_CODE | b64enc }}
+  HOTJAR_ID: {{ .HOTJAR_ID | b64enc }}
+  AWS_ACCESS_KEY: {{ .AWS_ACCESS_KEY | b64enc }}
+  AWS_SECRET_KEY: {{ .AWS_SECRET_KEY | b64enc }}
+  S3_BUCKET: {{ .S3_BUCKET | b64enc }}
+  NATIONBUILDER_TOKEN: {{ .NATIONBUILDER_TOKEN | b64enc }}
+  NATIONBUILDER_CLIENT_ID: {{ .NATIONBUILDER_CLIENT_ID | b64enc }}
+  NATIONBUILDER_CLIENT_SECRET: {{ .NATIONBUILDER_CLIENT_SECRET | b64enc }}
+  MAILCHIMP_API_KEY: {{ .MAILCHIMP_API_KEY | b64enc }}
+  MAILCHIMP_ANNOUNCE_ID: {{ .MAILCHIMP_ANNOUNCE_ID | b64enc }}
+  MAILCHIMP_WEBHOOK_KEY: {{ .MAILCHIMP_WEBHOOK_KEY | b64enc }}
+  RECAPTCHA_PUBLIC_KEY: {{ .RECAPTCHA_PUBLIC_KEY | b64enc }}
+  RECAPTCHA_PRIVATE_KEY: {{ .RECAPTCHA_PRIVATE_KEY | b64enc }}
+  SIMPLE_JWT_SIGNING_KEY: {{ .SIMPLE_JWT_SIGNING_KEY | b64enc }}
+  MOBILE_APP_KEY: {{ .MOBILE_APP_KEY | b64enc }}
```

Comment:
I think for ease of maintenance and setting up environments where a subset of these values differ, it might be a good idea to split these out into a set of secrets, each of which covers a different configuration aspect.  That way someone would be able to update just AWS or database via the values file without also having to also populate all the other fields via values.  Or we could implement a default-override fallthrough set of imports.  @moshloop what do you think?

Reply:
Waiting on this.